Snakefile

```diff
@@ -1,60 +1,58 @@
 from snakemake.exceptions import WorkflowError
 import os
-from typing import List, Dict
-import re
 import json
-import pandas as pd
-from mavis_config import validate_config
+from mavis_config import (
+    count_total_rows,
+    get_library_inputs,
+    get_singularity_bindings,
+    guess_total_batches,
+    validate_config,
+)
 from mavis_config.constants import SUBCOMMAND
 
-CONTAINER = 'bcgsc/mavis:latest'
+# env variable mainly for CI/CD
+CONTAINER = os.environ.get('SNAKEMAKE_CONTAINER', 'docker://bcgsc/mavis:latest')
+MAX_TIME = 57600
+DEFAULT_MEMORY_MB = 16000
+
+
+if 'output_dir' not in config:
+    raise WorkflowError('output_dir is a required property of the configfile')
+
 
 def output_dir(*paths):
     return os.path.join(config['output_dir'], *paths)
 
+
 INITIALIZED_CONFIG = output_dir('config.json')
+LOG_DIR = output_dir('logs')
 
+# external schedulers will not create the log dir if it does not already exist
+if not os.path.exists(LOG_DIR):
+    os.makedirs(LOG_DIR, exist_ok=True)
 
 try:
     validate_config(config, stage=SUBCOMMAND.SETUP)
 except Exception as err:
     short_msg = ' '.join(str(err).split('\n')[:2]) # these can get super long
     raise WorkflowError(short_msg)
 
+# ADD bindings for singularity
+workflow.singularity_args = f'-B {",".join(get_singularity_bindings(config))}'
```

Comment:
Would this work incase someone gave snakemake commanline arguments?
```
if not workflow.singularity_args:
    workflow.singularity_args = f'-B {",".join(get_singularity_bindings(config))}'
```

Reply:
not if they passed it the singularity args. But if they are advanced enough to customize their singularity args they can probably also figure out the bindings